### PR TITLE
Make contributing easier

### PR DESCRIPTION
**Pull request by Estus-Dev**
I've added a [Contributing section to the README](./README.md#Contributing), detailing how to add new programs to the list without being rejected by CI.

I've also fixed a bug that's bitten me both times I've contributed in the past, where `scripts/update-chip8-archive.js` would inject invalid colors into existing entries, causing schema validation to fail.

This was caused by the Chip8Archive's `programs.json` file containing programs where no colors were specified.

In future I'd like to automatically create _(draft)_ PRs for updates to the Chip8Archive, at which point we could remove that source of friction from new manual PRs and simplify the README I just added.

### CODE REVIEW

File: README.md
Comment (Timendus) [inline README.md]: I don't think this is necessary. The Github CI "only" checks to see if everything is formatted correctly, and passes the schema. It does this:

https://github.com/chip-8/chip-8-database/blob/bff63251ff9d0c95b2aab4c4408f92fb4a8d23f9/.github/workflows/test.yaml#L29-L31

Which do:

https://github.com/chip-8/chip-8-database/blob/bff63251ff9d0c95b2aab4c4408f92fb4a8d23f9/package.json#L7-L8

So that's just a Prettier check, and invoking the `validate.js` script, which does this, according to the comment:

https://github.com/chip-8/chip-8-database/blob/bff63251ff9d0c95b2aab4c4408f92fb4a8d23f9/scripts/validate.js#L1-L3

I wouldn't want to burden new contributors with having to update the CHIP-8 Archive before they can add their own programs.

Edit: which should also lessen the issue of the colours getting messed up by the import script. That's a script to be ran by maintainers, not random contributors, IMO. But still great to fix the issue.

File: README.md
Comment (Timendus) [inline README.md]: Maybe just say:

> Once you're done making your changes, run `npm run format` and make sure `npm test` passes before committing.

**Review by Timendus — CHANGES_REQUESTED** [review]
Looking great. One big remark below.

File: README.md
Comment (Estus-Dev) [reply README.md]: IMO it also would help new contributors if `npm test` is all you need to ensure you pass CI. Could we rename `scripts.test` to `scripts.validate-schema` to free it up for `npm run validate-schema && npm run check-formatting`?

File: README.md
Comment (Estus-Dev) [reply README.md]: I also don't like that being there and would be glad to remove it. In fact I was planning on adding a weekly CI run to create a PR a la dependabot and removing it then. That PR would need to be a draft for manual review of platforms/etc... but it'd be a nice prompt to action.

The reason I put it there was because I had that issue both times when contributing: because I ran `npm start` to generate hashes. That's the default command I try when I know there are `package.json` scripts to be run.

What do you think about about moving `scripts.start` to `scripts.update`, and not having an `npm start`?

Comment (Estus-Dev) [reply README.md]: Or alternatively having `npm start` run `npm run generate-hashes-file && npm run format`?

File: README.md
Comment (Estus-Dev) [reply README.md]: > run `npm run format` and make sure `npm test` passes before committing.

As an aside: If I shorten it to those two commands, then the SHA-1 file won't be updated.

File: README.md
Comment (Timendus) [reply README.md]: That sounds like a great idea 👍

File: README.md
Comment (Timendus) [reply README.md]: Ah, yes. It also needs a `generate-hashes-file` in there. Good catch. I'm fine with moving a couple of commands around if that makes things easier for newcomers 💪

**Comment by Estus-Dev** [comment]
EDIT: Didn't refresh the page before commenting.

I figured it'd be easier to make the changes I proposed than to get signoff first and then make them and then have you review them again.

In particular, I moved the updating section to the bottom of the page because it's still useful to have documented, but I removed the incorrect info about CI.

I changed the commands around as I suggested, so `npm start` and `npm test` are all a new contributor needs to know, while leaving `npm run update` for pulling from upstream.

**Pull request merged**
